src/tasks/task_list.hpp

```diff
@@ -89,6 +164,52 @@ class TaskList {
                          [=]() mutable -> TaskStatus { return (obj->*func)(args...); });
   }
 
+  template <class F, class... Args>
+  TaskID AddIterativeTask(const TaskType &type, const std::string &label,
```

Comment:
Could `AddIterativeTask` be an overload of just `AddTask`? Or would the templating not like that?

Reply:
I didn't try.  It might be OK, but I kind of liked the explicitness of calling out these kinds of tasks as iterative.  I'd be curious if anyone else has an opinion on this.  I'd be fine with being overruled and try the overload.

Reply:
We could also add the type of task as a tag. e.g.,
```C++
AddTask(TaskType::Iterative,...);
AddTask(TaskType::Normal,...);
//etc
```

Reply:
But on the other hand that's API breaking so NVM